NineAnimator/Controllers/HomeController.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  This file is part of the NineAnimator project.
+//
+//  Copyright © 2018 Marcus Zhou. All rights reserved.
+//
+//  NineAnimator is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  NineAnimator is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with NineAnimator.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import HomeKit
+
+/**
+ Manages HomeKit integration
+ 
+ Reference to `HomeController` is maintained in the `AppDelegate`.
+ HomeController operates by tracking the updating playback
+ progress notification.
+ */
+class HomeController: NSObject, HMHomeManagerDelegate {
+    static weak var shared: HomeController?
+    
+    private var _isReady: Bool = false
+    
+    var isReady: Bool {
+        _ = manager
+        return _isReady
+    }
+    
+    private lazy var manager: HMHomeManager = {
+        // Creating this object is known to trigger main thread checker
+        // But it doesn't seem to affect anything
+        Log.info("Initializing HMHomeManager. An error might be thrown, which doesn't seem to affect anything.")
+        let manager = HMHomeManager()
+        manager.delegate = self
+        return manager
+    }()
+    
+    override init() {
+        super.init()
+        
+        if HomeController.shared != nil { Log.error("HomeController is initialized twice.") }
```

Comment:
All static let “variables” are initialized only once. Why did you choose this approach?